shim/src/main.rs

```diff
@@ -124,36 +177,162 @@ enum RepoMode {
+/// If a command has a help flag passed, print help and return true.
+/// Otherwise, return false.
 ///
 /// # Arguments
 ///
-/// * `clap_args`: Parsed arguments from clap
-/// * `args`: Raw un-parsed arguments
+/// * `command`: The parsed command.
 ///
-/// returns: Result<i32, Error>
-fn run_current_turbo(clap_args: Args, args: Vec<String>) -> Result<i32> {
-    if let Some(Command::Bin) = clap_args.command {
-        commands::bin::run()?;
-        return Ok(0);
+/// returns: Result<bool, Error>
+fn try_run_help(command: &Command) -> Result<bool> {
+    let (help, command_name) = match command {
+        Command::Bin { help, .. } => (help, "bin"),
+        Command::Completion { help, .. } => (help, "completion"),
+        Command::Daemon { help, .. } => (help, "daemon"),
+        Command::Help { help, .. } => (help, "help"),
+        Command::Link { help, .. } => (help, "link"),
+        Command::Login { help, .. } => (help, "login"),
+        Command::Logout { help, .. } => (help, "logout"),
+        Command::Prune { help, .. } => (help, "prune"),
+        Command::Run { help, .. } => (help, "run"),
+        Command::Unlink { help, .. } => (help, "unlink"),
+    };
+
+    if *help {
+        Args::command()
+            .find_subcommand_mut(command_name)
+            .expect("Could not find subcommand")
+            .print_long_help()?;
+        Ok(true)
+    } else {
+        Ok(false)
     }
+}
 
-    let mut args = args
-        .into_iter()
-        .map(|s| {
-            let c_string = CString::new(s)?;
-            Ok(c_string.into_raw())
-        })
-        .collect::<Result<Vec<*mut c_char>>>()?;
-    args.shrink_to_fit();
-    let argc: c_int = args.len() as c_int;
-    let argv = args.as_mut_ptr();
-    let exit_code = unsafe { nativeRunWithArgs(argc, argv) };
-    Ok(exit_code.try_into().unwrap())
+impl TurboState {
+    /// Runs the Go code linked in current binary.
+    ///
+    /// # Arguments
+    ///
+    /// * `args`: Arguments for turbo
+    ///
+    /// returns: Result<i32, Error>
+    fn run_current_turbo(self) -> Result<i32> {
+        match self.parsed_args.command {
+            Some(Command::Bin { .. }) => {
+                commands::bin::run()?;
+                Ok(0)
+            }
+            Some(Command::Link { .. })
+            | Some(Command::Login { .. })
+            | Some(Command::Logout { .. })
+            | Some(Command::Unlink { .. }) => {
+                let exit_code = unsafe { nativeRunWithTurboState(self.try_into()?) };
+                Ok(exit_code.try_into()?)
+            }
+            _ => {
+                let mut args = self
+                    .raw_args
+                    .iter()
+                    .map(|s| {
+                        let c_string = CString::new(s.as_str())?;
+                        Ok(c_string.into_raw())
+                    })
+                    .collect::<Result<Vec<*mut c_char>>>()?;
+                args.shrink_to_fit();
```

Comment:
Honestly this is some code either written by Greg or used from a tutorial. But I'd bet that yes, it probably is because there's a chance that the system allocator uses some large alignment number, or the vector doubles on growth (or 1.6x, there's some "optimal" constant). I'll add a comment